File: other_impls/sieve.c
#include <stdio.h>
#include <assert.h>
#include <string.h>

#define MAX_PRIME 100000

void sieve(int max_n, short primes[MAX_PRIME]) {
  short zero = 0;
  memset(primes, zero, MAX_PRIME);

  for(int n=2; n<max_n; n++) {
    int current_n = n + n;
    while(current_n < max_n) {
      primes[current_n] = 1;
      current_n += n;
    }
  }
}

int main(void) {
  int max_n = 100;
  short primes[MAX_PRIME];
  assert(max_n < MAX_PRIME);
  
  sieve(100, primes);
  
  for(int i=2; i<max_n; i++) {
    if(primes[i] == 0) {
      printf("%i, ", i);
    }
  }
  printf("\n");
  
  return 0;
}

File: other_impls/sieve.lua
function sieve(max_n)
   number_map = {}
   for n = 2, max_n + 1 do
      number_map[n] = 0
   end

   for n, value in pairs(number_map) do
      current_n = n + n
      while current_n < max_n do
         number_map[current_n] = number_map[current_n] + 1
         current_n = current_n + n
      end
   end

   primes = {}
   for n, value in pairs(number_map) do
      if value == 0 then
         table.insert(primes, n)
      end
   end
   return primes
end

for index, prime in pairs(sieve(100)) do
   print(prime)
end

print(table.concat(sieve(100), ", "))